common/include/pcl/point_types.h

```diff
@@ -790,12 +790,33 @@ namespace pcl
                                                             has_field<PointT, boost::mpl::_2> > >::type
     { };
 
+    /** \brief Traits defined for ease of use with fields
+     * 
+     * has_<fields to be detected>: struct with `value` datamember defined at compiletime
+     * has_<fields to be detected>_v: constexpr boolean
+     * Has<Fields to be detected>: concept modelling name alias for `enable_if`
+     */
+
+    /** Metafunction to check if a given point type has x, y, and z fields. */
+    template <typename PointT>
+    struct has_xy : has_all_fields<PointT, boost::mpl::vector<pcl::fields::x,
```

Comment:
This implementation will return a positive result for `PointXYZ`. I don't think this is what we want, so probably should additionally check that `z` field is *not* present.

Reply:
We are just checking if `x` and `y` are present. Prohibiting on `z` might not be the intent a user might parse

Reply:
Well, depends on the use-case indeed. What comes to my mind is a specialization of some template function/class for 2D points. In that case, we'd want to "activate" a specialization just for `PointXY`, not for `PointXYZ` points. On the other hand, checking `isXYFinite<PointXYZ>()` is probably a valid use-case, so your implementation also makes sense.

Reply:
> we'd want to "activate" a specialization just for PointXY

For this I can think of a wording along the lines of `is_point_2d`. `has_xy` isn't exclusive, but `is_2d` or `is_point_2d` are excluding 1D and 3D cases among others